datahugger/services.py

```diff
@@ -445,3 +451,113 @@ class B2shareDataset(DatasetDownloader):
     ATTR_SIZE_JSONPATH = "size"
     ATTR_HASH_JSONPATH = "checksum"
     ATTR_HASH_TYPE_VALUE = "md5"
+
+
+class YodaDataset(DatasetDownloader):
+    """Downloader for Yoda repositories."""
+
+    def _get_contents_url(self):
+        """Resolve the root folder of the contents of a Yoda data package."""
+        url_to_use = (
+            self.resource if isinstance(self.resource, str) else self.resource.resolve()
+        )
+        res = requests.get(url_to_use)
+        res.raise_for_status()
+        soup = BeautifulSoup(res.content, "html.parser")
+        contents_link = soup.find("a", string=re.compile(r"View contents"))
+        return contents_link.get("href") + "/original" if contents_link else None
+
+    @property
+    def files(self):
+        if not hasattr(self, "_files"):
+            self._requests_cache_file = tempfile.NamedTemporaryFile(delete=False)
+            requests_cache.install_cache(self._requests_cache_file.name)
+            self._files = self._harvest_files()
+            self._cleanup_requests_cache()
+        return self._files
```

Comment:
I wonder why you cache this. It makes sense, but is there a reason to do this for Yoda specifically? Or should we implement this feature for all services in a generic way?

Reply:
The original intent of this part of the code was to cache DNS query responses. In case of Yoda we need to request every file in the data set individually (rather than say, just requesting a single zip file that contains all the files). This can result in significant overhead for name resolution. Apart from that, flaky DNS servers can result in failures to harvest all files.

However, this solution ultimately involves monkey patching the `requests` module (or one of the lower-level modules), which can potentially interfere with other software that depends on datahugger. The implementation also didn't help (that much) with improving performance.

After reconsidering, I have removed this part of the code.